server/server.go

```diff
@@ -328,17 +320,10 @@ func (t *txpoolHub) GetBalance(root types.Hash, addr types.Address) (*big.Int, e
 	if err != nil {
 		return nil, fmt.Errorf("unable to get snapshot for root, %w", err)
 	}
-
-	result, ok := snap.Get(keccak.Keccak256(nil, addr.Bytes()))
-	if !ok {
+	account, err := snap.GetAccount(addr)
+	if err != nil {
 		return big.NewInt(0), nil
```

Comment:
Would you like to propagate error here perhaps, as `GetBalance` function signature implies returning an error?